File: setup/src/main/java/es/codeurjc/daw/library/dto/TeamDTO.java
package es.codeurjc.daw.library.dto;
import java.util.List;

public record TeamDTO(
    Long id, 
    String username, 
    String email, 
    String teamName, 
    boolean hasImage,
    List<PlayerBasicDTO> players,
    List<TournamentBasicDTO> tournaments
) {}

File: frontend/exam0_minimal/app/dtos/TeamDTO.ts
export interface PlayerDTO {
    id: number;
    name: string;
    position: string;
    jerseyNumber: number;
    goals: number;
    assists: number;
    hasImage: boolean;
}

export interface TournamentBasicDTO {
    id: number;
    name: string;
    state: string;
    status: string;
    type: string;
}

export interface TeamDTO {
    id: number;
    username: string;
    teamName: string;
    email: string;
    hasImage: boolean;
    players: PlayerDTO[];
    tournaments: TournamentBasicDTO[];
}
